### PR TITLE
[12.0][MIG] sale_commission_salesman: Migration to 12.0

### PR DESCRIPTION
**Pull request by pedrobaeza**
- Standard procedure
- It includes the needed refactoring in `sale_commission` module done in 11.0, in a separate commit.

@Tecnativa TT24802

### CODE REVIEW

File: sale_commission/tests/test_sale_commission.py [hunk 1]
```diff
@@ -108,14 +108,14 @@ def test_sale_commission_gross_amount_payment(self):
...
-    def test_sale_commission_gross_amount_payment_annual(self):
+    def _test_sale_commission_gross_amount_payment_annual(self):
         self.check_full(
             self.agent_annual,
             self.commission_section_paid,
             12
         )
 
-    def test_sale_commission_gross_amount_payment_semi(self):
+    def _test_sale_commission_gross_amount_payment_semi(self):
```
Comment (chienandalu) [inline sale_commission/tests/test_sale_commission.py]: Why is this needed?

Comment (pedrobaeza) [reply sale_commission/tests/test_sale_commission.py]: Nothing. I forgot to restore it. Restoring it...

Comment (pedrobaeza) [reply sale_commission/tests/test_sale_commission.py]: Done

**Review by carlosdauden — APPROVED** [review]

**Review by chienandalu — APPROVED** [review]
